internal/config/reloader/remote.go

```diff
@@ -0,0 +1,203 @@
+package reloader
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/r3labs/diff/v3"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+
+	"github.com/kubeshop/botkube/internal/config/remote"
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+// DeploymentClient defines GraphQL client.
+type DeploymentClient interface {
+	GetResourceVersion(ctx context.Context) (int, error)
+	GetConfigWithResourceVersion(ctx context.Context) (remote.Deployment, error)
+}
+
+// NewRemote returns new ConfigUpdater.
+func NewRemote(log logrus.FieldLogger, deployCli DeploymentClient, restarter *Restarter, cfg config.Config, cfgVer int, resVerHolders ...ResourceVersionHolder) *RemoteConfigReloader {
+	return &RemoteConfigReloader{
+		log:           log,
+		currentCfg:    cfg,
+		resVersion:    cfgVer,
+		interval:      cfg.ConfigWatcher.Remote.PollInterval,
+		deployCli:     deployCli,
+		resVerHolders: resVerHolders,
+		restarter:     restarter,
+	}
+}
+
+type RemoteConfigReloader struct {
```

Comment:
not sure what's your approach on this PR but it's quite random when it comes to commenting exported struct/methods 😄

Reply:
Whenever I see I didn't put a comment, I try to do it 😄 Will fix it - I think we have the linter errors silenced for internal 🤔 (the comments-related check)